tools/scylla-nodetool.cc

```diff
@@ -1428,6 +1441,616 @@ void stop_operation(scylla_rest_client& client, const bpo::variables_map& vm) {
+    {}
+
+    bool operator()(const sstring& ks, const sstring& cf) const {
+        if (belongs_to_filter(ks, cf)) {
+            return _mode == mode::inclusive;
+        } else {
+            return _mode == mode::exclusive;
+        }
+    }
+};
+
+void table_stats_print_plain(scylla_rest_client& client,
+                             const table_filter& is_included,
+                             bool human_readable) {
+    fmt::print("Total number of tables: {}\n", client.get("/column_family/").GetArray().Size());
+    fmt::print("----------------\n");
+    for (auto& [keyspace_name, table_names] : get_ks_to_cfs(client)) {
+        keyspace_stats keyspace;
+        for (auto& table_name : table_names) {
+            keyspace.add(table_metrics(client, keyspace_name, table_name),
+                         is_included(keyspace_name, table_name));
+        }
+        fmt::print("Keyspace : {}\n", keyspace_name);
+        fmt::print("\tRead Count: {}\n", keyspace.read_count);
+        fmt::print("\tRead Latency: {:.15E} ms\n", keyspace.read_latency());
+        fmt::print("\tWrite Count: {}\n", keyspace.write_count);
+        fmt::print("\tWrite Latency: {:.15E} ms\n", keyspace.write_latency());
+        fmt::print("\tPending Flushes: {}\n", keyspace.pending_flushes);
+
+        for (auto& table : keyspace.tables) {
+            table.print(human_readable);
+        }
+        fmt::print("----------------\n");
+    }
+}
+
+void table_stats_print_json(scylla_rest_client& client,
+                            const table_filter& is_included,
+                            bool human_readable) {
+    // {
+    //     'total_number_of_tables': ...,
+    //     'keyspace1' : {
+    //         'read_latency': ...,
+    //         # ...
+    //         'tables': {
+    //             'cf1': {
+    //                 'sstables_in_each_level': ...
+    //                 # ...
+    //             },
+    //             # ...
+    //         }
+    //     }
+    //     # ...
+    // }
+    rjson::streaming_writer writer;
+    writer.StartObject();
+    writer.Key("total_number_of_tables");
+    writer.Write(client.get("/column_family/").GetArray().Size());
+    for (auto& [keyspace_name, table_names] : get_ks_to_cfs(client)) {
+        keyspace_stats keyspace;
+        for (auto& table_name : table_names) {
+            keyspace.add(table_metrics(client, keyspace_name, table_name),
+                         is_included(keyspace_name, table_name));
+        }
+        writer.Key(keyspace_name);
+        writer.StartObject();
+        dump_map_to_json(writer, keyspace.to_map());
+        writer.Key("tables");
+        writer.StartObject();
+        for (auto& table : keyspace.tables) {
+            writer.Key(table.name());
+            writer.StartObject();
+            dump_map_to_json(writer, table.to_map(human_readable));
+            writer.EndObject();
+        }
+        writer.EndObject();
+        writer.EndObject();
+    }
+    writer.EndObject();
+}
```

Comment:
yeah, this is not pretty. will try to consolidate yaml and json builder in a follow-up change.